### PR TITLE
CI: build with OTP 23

### PR DESCRIPTION
**Pull request by mattludwigs**

### CODE REVIEW

**Comment by mattludwigs** [comment]
Yep, I have done that on just about every one of these types of PRs.

File: .circleci/config.yml [hunk 1]
```diff
@@ -33,6 +33,33 @@ defaults: &defaults
   working_directory: ~/repo
 
 jobs:
+  build_elixir_1_10_otp_23:
+    docker:
+      - image: erlang:23.0
```
Comment (fhunleth) [inline .circleci/config.yml:38]:
```suggestion
      - image: erlang:23.0.1
```
This avoids a random compiler hang.

File: .circleci/config.yml [hunk 2]
```diff
@@ -33,6 +33,33 @@ defaults: &defaults
...
+      - run: mix dialyzer
+      - save_cache:
+          key: v1-mix-cache-{{ checksum "mix.lock" }}
+          paths:
+            - _build
+            - deps
+
   build_elixir_1_10_otp_22:
     docker:
       - image: erlang:22.3
```
Comment (fhunleth) [inline .circleci/config.yml:65]:
```suggestion
      - image: erlang:22.3.4.1
```

File: mix.exs [hunk 3]
```diff
@@ -17,7 +17,8 @@ defmodule VintageNetWizard.MixProject do
       docs: docs(),
       aliases: [docs: ["docs", &copy_images/1]],
       package: package(),
-      description: description()
+      description: description(),
+      preferred_cli_env: [docs: :docs, "hex.publish": :docs]
```
Comment (fhunleth) [inline mix.exs:21]:
```suggestion
      preferred_cli_env: [docs: :docs, "hex.build": :docs, "hex.publish": :docs, "coveralls.circle": :test]
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -33,6 +33,33 @@ defaults: &defaults
...
+    steps:
+      - checkout
+      - <<: *install_system_deps
+      - <<: *install_elixir
+      - <<: *install_hex_rebar
+      - restore_cache:
+          keys:
+            - v1-mix-cache-{{ checksum "mix.lock" }}
+      - run: mix deps.get
+      - run: MIX_ENV=test mix coveralls.circle
```
Comment (fhunleth) [inline .circleci/config.yml:52]:
```suggestion
      - run: mix coveralls.circle
```

**Pull request merged**